packages/ra-core/src/sideEffect/useNotify.ts

```diff
@@ -37,7 +37,7 @@ const useNotify = () => {
         ) => {
             if (typeof type === 'string' || !type) {
                 warning(
-                    true,
+                    typeof type === 'string',
```

Comment:
I don't get this fix. If you call `notify` with only one argument then `type` isn't a string as it would be undefined, right?

Reply:
Sure, since `type` will be `undefined`, the warning will not show, since now the truthy condition is set to be thrown when `type` is passed as a string

Reply:
Yes, but we check this condition at L38 already

Reply:
If you call `notify` with only one argument (the message), then the condition at L38 is false and we won't each the warning, right?

Reply:
My idea is to not show a warning if `type` prop is not passed. Do you have any suggestions to improve this?

Reply:
np :)

Reply:
I was incorrectly trying to avoid warnings from within react-admin tests, but that is another pr

Reply:
Yes, we have to migrate all usages of `useNotify` in our tests

Reply:
Hold on,  if I call `notify` with only one argument (the message), the condition in line 38 will be true, because `!type` is true, and I will see a warning, right?

Reply:
Right, let's just remove the second condition then

Reply:
Ok